### PR TITLE
Reference+Selection Updates [Final Batch] 

### PR DESCRIPTION
**Pull request by nasbench**
- Updated all* Github references to permalinks
- Update selection to be more conform 
- Update indentation to be more conform
- Deleted DNS rules (71ba22cb-8a01-42e2-a6dd-5bf9b547498f + 7c4cf8e0-1362-48b2-a512-b606d2065d7d + ed785237-70fa-46f3-83b6-d264d1dc6eb4) and merged them into one (4d07b1f4-cb00-4470-b9f8-b0191d48ff52) because they are describing the same behaviour
- Added 3 new rules
  - 0adc67e0-a68f-4ffd-9c43-28905aad5d6a (Detect default KOH named pipes)
  - 21f9162c-5f5d-4b01-89a8-b705bd7d10ab (Detect importing PowerShell module from susp locations)
  - bd33d2aa-497e-4651-9893-5c5364646595 (Detect PowerShell tunneling via function use)

### CODE REVIEW

**Comment by Neo23x0** [comment]
I'll only check the 3 new rules - I cannot review all the other changes

File: rules/linux/process_creation/proc_creation_lnx_security_software_discovery.yml [hunk 1]
```diff
@@ -4,29 +4,29 @@ status: test
...
-      - 'nessusd'        # nessus vulnerability scanner
-      - 'td-agent'       # fluentd log shipper
-      - 'packetbeat'     # elastic network logger/shipper
-      - 'filebeat'       # elastic log file shipper
-      - 'auditbeat'      # elastic auditing agent/log shipper
-      - 'osqueryd'       # facebook osquery
-      - 'cbagentd'       # carbon black
-      - 'falcond'        # crowdstrike falcon
-  condition: grep_execution and security_services_and_processes
+    grep_execution:
```
Comment (frack113) [inline rules/linux/process_creation/proc_creation_lnx_security_software_discovery.yml:14]:
```yaml
    selection:
        Image|endswith: '/grep'
        CommandLine|contains:
            - 'nessusd'        # nessus vulnerability scanner
            - 'td-agent'       # fluentd log shipper
            - 'packetbeat'     # elastic network logger/shipper
            - 'filebeat'       # elastic log file shipper
            - 'auditbeat'      # elastic auditing agent/log shipper
            - 'osqueryd'       # facebook osquery
            - 'cbagentd'       # carbon black
            - 'falcond'        # crowdstrike falcon
    condition: selection

File: rules/web/web_cve_2021_41773_apache_path_traversal.yml [hunk 2]
```diff
@@ -4,30 +4,30 @@ status: experimental
...
     - https://twitter.com/h4x0r_dz/status/1445401960371429381
-    - https://github.com/projectdiscovery/nuclei-templates/blob/master/cves/2021/CVE-2021-41773.yaml
+    - https://github.com/projectdiscovery/nuclei-templates/blob/9d2889356eebba661c8407038e430759dfd4ec31/cves/2021/CVE-2021-41773.yaml
     - https://twitter.com/bl4sty/status/1445462677824761878
 logsource:
   category: webserver
 detection:
-    selection:
-        c-uri|contains: 
+    selection_uri:
```
Comment (frack113) [inline rules/web/web_cve_2021_41773_apache_path_traversal.yml:18]:
```yaml
    selection:
        c-uri|contains:
            - '/cgi-bin/.%2e/'
            - '/icons/.%2e/'
            - '/cgi-bin/.%%32%65/'
            - '/icons/.%%32%65/'
            - '/cgi-bin/.%%%25%33'
            - '/icons/.%%%25%33'
        sc-status:
            - 200
            - 301
    condition: selection
```

**Review by frack113 — COMMENTED** [review]
100 of 427

**Review by frack113 — APPROVED** [review]

**Pull request merged**